### ISSUE TITLE
__init__.py:21-23: Add SecuredView, with accompanied...

### ISSUE THREAD

**Issue by 0pdd** [timelesslounge/timelessis#179]
The puzzle `22-9cc1d6a5` from #22 has to be resolved: 

https://github.com/timelesslounge/timelessis/blob/ae4c725b13e7afb9f349eb20f55e5a2d22cd4ebb/timeless/access_control/__init__.py#L21-L23

The puzzle was created by Vatavuk on 09-Feb-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#179](https://github.com/timelesslounge/timelessis/issues/179) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#179](https://github.com/timelesslounge/timelessis/issues/179) assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/09568e15-0d33-4558-bab5-b2b54b1e170b); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by amihaiemil** [comment]
@0crat wait for #183 . SecuredView doesn't seem to work properly yet. Trying to use it in an existing view class and getting the exception: ``TypeError: get() takes 2 positional arguments but 4 were given``.

**Comment by 0crat** [comment]
> @0crat wait for #183 . SecuredView doesn't seem to work properly yet. Trying to use it in an exis... ([here](https://github.com/timelesslounge/timelessis/issues/179#issuecomment-462057991))

@amihaiemil The impediment for [#179](https://github.com/timelesslounge/timelessis/issues/179) was registered successfully by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by emilianodellacasa** [comment]
@vatavuk what's your opinion?

**Comment by Vatavuk** [comment]
@amihaiemil @emilianodellacasa Small error in `SecuredView.dispatch_request`:
Instead of : 
```python 
super(SecuredView, self).dispatch_request(self, args, kwargs)
``` 
we should invoke method without `self`, like this:
```python 
super(SecuredView, self).dispatch_request(args, kwargs)
```

**Comment by emilianodellacasa** [comment]
@Vatavuk thank you so much for the explenation

**Comment by 0pdd** [comment]
The puzzle `22-9cc1d6a5` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd the puzzle [#278](https://github.com/timelesslounge/timelessis/issues/278) is still not solved.

**Comment by 0crat** [comment]
Job [#179](https://github.com/timelesslounge/timelessis/issues/179) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
The job [#179](https://github.com/timelesslounge/timelessis/issues/179) is now out of scope

**Comment by amihaiemil** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/179#issuecomment-464487975))

@amihaiemil This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#179](https://github.com/timelesslounge/timelessis/issues/179) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 3min
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will receive 30 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23179%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by amihaiemil** [comment]
@ypshenychka can you pls make the QA review here? 0crat forgot to mention you again :)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#473](https://github.com/timelesslounge/timelessis/issues/473), [#474](https://github.com/timelesslounge/timelessis/issues/474) are still not solved; solved: [#278](https://github.com/timelesslounge/timelessis/issues/278).
